app/routes/__init__.py

```diff
@@ -137,6 +146,48 @@ async def raster_tile_cache_version_dependency(
     return dataset, version
 
 
+async def cog_asset_dependency(
+    request: Request,
+    dataset: Optional[COGDatasets] = Query(None, description=COGDatasets.__doc__),
+    version: Optional[str] = Query(
+        None, description="Data API dataset version.", regex=VERSION_REGEX_NO_LATEST
+    ),
+    name: Optional[str] = Query(
+        "default", description="Name of the COG asset to display"
+    ),
+    url: Optional[str] = Query(
+        None,
+        description="Dataset path. This needs to be set if `dataset` and `version` query parameters for a Data API dataset are not set.",
+    ),
+) -> Optional[str]:
+
+    if dataset is None and version is None and url is None:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            "Need to pass either `url` or `dataset` and `version` pair for Data API dataset.",
+        )
+
+    if dataset and version and url:
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST,
+            "Need to pass either `url` or `dataset` and `version` pair, not both.",
+        )
+
+    if dataset and version:
+        folder: str = (
+            f"s3://{DATA_LAKE_BUCKET}/{dataset}/{version}/raster/epsg-4326/cog"
```

Comment:
Ah, are you saying it's hard to include projection in the URI because then you need to look it up here?

Reply:
Yes. Also using this directory structure feels like forcing titiler to fit the existing pipeline. For example, if someone wants to create a file in UTM, it feels redundant to create a directory to just store that file, and it would add more effort to navigate the tree, etc.